src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.cpp

```diff
@@ -43,330 +43,386 @@
 
-	// Initialize to something such that the drone at least points at the target, even if it's the wrong angle for the perspective.
-	// The drone will move into position as soon as the target starts moving and its heading becomes known.
-	Vector2f current_drone_heading_2d{cosf(_yaw), -sinf(_yaw)};
+	_yaw_setpoint = _yaw;
+	_yawspeed_setpoint = NAN;
 
-	if (PX4_ISFINITE(current_drone_heading_2d(0)) && PX4_ISFINITE(current_drone_heading_2d(1))) {
-		_offset_vector_filter.reset(current_drone_heading_2d);
+	// Update the internally tracked Follow Target characteristics
+	_follow_angle_rad = convertFollowPerspectiveToRadians((kFollowPerspective)_param_flw_tgt_fp.get());
+	_follow_distance = _param_flw_tgt_dst.get();
+	_follow_height = _param_flw_tgt_ht.get();
 
-	} else {
-		_offset_vector_filter.reset(Vector2f(1.0f, 0.0f));
-	}
+	// Reset the orbit angle trajectory generator and set maximum angular acceleration and rate
+	_orbit_angle_traj_generator.reset(0.f, 0.f, 0.f);
 
-	_yawspeed_setpoint = 0.f;
+	// Save the home position z value to enable relative altitude setpoints to arming position (home) altitude
+	if (_sub_home_position.get().valid_alt) {
+		_home_position_z = _sub_home_position.get().z;
+	}
+	else {
+		ret = false; // Don't activate Follow Target task if home position is not valid
+	}
 
 	return ret;
 }
 
-Vector2f FlightTaskAutoFollowTarget::calculate_offset_vector_filtered(Vector3f vel_ned_est)
-{
-	if (_param_nav_ft_fs.get() == FOLLOW_PERSPECTIVE_NONE) {
-		// NOTE: Switching between NONE any any other setting currently causes a jump in the setpoints
-		_offset_vector_filter.reset(Vector2f{0, 0});
+void FlightTaskAutoFollowTarget::updateParams() {
+	// Copy previous param values to check if it changes after param update
+	const float follow_distance_prev = _param_flw_tgt_dst.get();
+	const float follow_height_prev = _param_flw_tgt_ht.get();
+	const kFollowPerspective follow_side_prev = (kFollowPerspective)_param_flw_tgt_fp.get();
 
-	} else {
-		// Define and rotate offset vector based on follow-me perspective setting
-		const float new_follow_angle_rad = math::radians(update_follow_me_angle_setting(_param_nav_ft_fs.get()));
+	// Call updateParams in parent class to update parameters from child up until ModuleParam base class
+	FlightTask::updateParams();
 
-		// Use shortest rotation to get to the new angle
-		// Example: if the current angle setting is 270, and the new angle setting is 0, it's
-		// faster to rotate to 360 rather than 0.
-		// Usually the controller would automatically take the shortest path. But here some trickery
-		// is necessary because the yaw angle is run through a low-pass filter.
-		if (_follow_angle_rad - new_follow_angle_rad  > M_PI_F) {
-			_follow_angle_rad = new_follow_angle_rad + M_TWOPI_F;
+	// Compare param values and if they have changed, update the properties
+	if (matrix::isEqualF(follow_distance_prev, _param_flw_tgt_dst.get())) {
```

Comment:
That statement doesn't do what the comment says...

Reply:
Oh it is definitely a bug!!! :palm_tree: How did I miss this!